ssz/src/main/java/org/ethereum/beacon/ssz/visitor/SSZSimpleHasher.java

```diff
@@ -72,10 +75,13 @@ public MerkleTrie visitComposite(SSZCompositeType type, Object rawValue,
     } else if ((type.getType() == LIST || type.getType() == VECTOR)
         && ((SSZListType) type).getElementType().getType() == BASIC) {
       SerializerResult sszSerializerResult = serializer.visitAny(type, rawValue);
-      SSZListAccessor listAccessor =
-          (SSZListAccessor) type.getAccessor().getInstanceAccessor(type.getTypeDescriptor());
-      BytesValue serialization =
-          listAccessor.removeListSize(rawValue, sszSerializerResult.getSerializedBody());
+      BytesValue serialization;
+      // Strip size bit in Bitlist
+      if (type.getType() == LIST && type instanceof SSZBitListType) {
```

Comment:
Why didn't you want to add a dedicated types `BITLIST` & `BITVECTOR` ?

Reply:
This will force to modify all other `== LIST` and `== VECTOR` comparisons to include `BITLIST` and `BITVECTOR` values as Enums doesn't have inheritance or smth like this. I've tried the easy way, check my next commit: https://github.com/harmony-dev/beacon-chain-java/pull/158/commits/e515237ac32e2fc37c9e210befa15458fad25c65